scs2-session-visualizer-jfx/src/main/java/us/ihmc/scs2/sessionVisualizer/jfx/session/mcap/MultiVideoDataReader.java

```diff
@@ -0,0 +1,75 @@
+package us.ihmc.scs2.sessionVisualizer.jfx.session.mcap;
+
+import org.bytedeco.ffmpeg.global.avutil;
+import org.bytedeco.javacv.FFmpegFrameGrabber;
+import us.ihmc.log.LogTools;
+import us.ihmc.scs2.sessionVisualizer.jfx.managers.BackgroundExecutorManager;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Future;
+
+public class MultiVideoDataReader
+{
+   static
+   {
+      // TODO: (AM) hack to suppress warnings from https://github.com/bytedeco/javacv/issues/780
+      avutil.av_log_set_level(avutil.AV_LOG_ERROR);
+   }
+
+   private final List<VideoDataReader> readers = new ArrayList<>();
+   private final BackgroundExecutorManager backgroundExecutorManager;
+   private Future<?> currentTask = null;
+
+   public MultiVideoDataReader(File dataDirectory, BackgroundExecutorManager backgroundExecutorManager)
+   {
+      this.backgroundExecutorManager = backgroundExecutorManager;
+      List<Path> videoFiles;
+      LogTools.info("Searching for videos in {}", dataDirectory.getAbsolutePath());
+      if (dataDirectory.isDirectory())
+      {
+         try
+         {
+            videoFiles = Files.walk(dataDirectory.toPath(), 1).filter(f -> f.toString().endsWith(".mp4") | f.toString().endsWith(".avi")).toList();
+            LogTools.info("Found video file(s): {}", videoFiles.toString());
+         }
+         catch (IOException e)
+         {
+            throw new RuntimeException(e);
+         }
+         for (int i = 0; i < videoFiles.size(); i++)
+         {
+            readers.add(new VideoDataReader(videoFiles.get(i).toFile()));
+         }
+      }
+   }
+
+   public void readVideoFrameNow(double time)
```

Comment:
I did this in seconds since different video containers can have different timebase values, and having the output timestamp calculation be calculated as `time * 1000000` allows us to be a bit more explicit about the fact that we are using the default value of `AV_TIME_BASE` as defined by FFMPEG.

Reply:
After re-reading the code, I find the time management here a little confusing.
The log is providing timestamp in nanoseconds saved as `long`. It is converted into seconds in the log reader and saved as `double`. The time in seconds is essentially only used to provide a human readable time unit that is used in the GUI. For any time comparison or operation, I prefer using the original timestamp in nanosecond provided by the log file.
The `VideoDataReader` now introduces a `long` timestamp that is in microsecond which is counter intuitive with the rest.
`VideoDataReader` is the one responsible for interfacing with ffmpeg, so I think the conversion should happen inside of that class, from outside, we shouldn't have to care for the fact that ffmpeg uses a different standard.
That makes sense?

Reply:
Ah I see, yeah the conversion should indeed by handled by the `VideoDataReader`, I'l make that change and just the nanosecond timestamp then